### PR TITLE
Feature 4099/include cql

### PR DESCRIPTION
**Pull request by sb-prateekkeerthi**
## MADiE PR

Jira Ticket: [MAT-4099](https://jira.cms.gov/browse/MAT-4099)
(Optional) Related Tickets:

### Summary
Including CQL editor in the CQL Library Create

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [ ] No extemporaneous files are included (i.e Complied files or testing results)
* [ ] This PR is in to the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependancies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/api/useElmTranslationServiceApi.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  externalErrors: any[];
+  library: ElmTranslationLibrary;
+};
+
+export class ElmTranslationServiceApi {
+  constructor(private baseUrl: string, private getAccessToken: () => string) {}
+
+  async translateCqlToElm(cql: string): Promise<ElmTranslation> {
+    if (this.baseUrl) {
+      const resp = await axios.put(`${this.baseUrl}/cql/translator/cql`, cql, {
```
Comment (adongare) [inline src/api/useElmTranslationServiceApi.ts]: this is an interesting URL

Comment (adongare) [inline src/api/useElmTranslationServiceApi.ts]: this is an interesting URL

**Review by adongare — DISMISSED** [review]

File: src/api/useElmTranslationServiceApi.ts (hunk 1 above)
Comment (sb-prateekkeerthi) [reply src/api/useElmTranslationServiceApi.ts]: This code is something related to next ticket, I forgot to push the code before working on the next ticket. I'll remove this file it is not related to this ticket

Comment (sb-prateekkeerthi) [reply src/api/useElmTranslationServiceApi.ts]: This code is something related to next ticket, I forgot to push the code before working on the next ticket. I'll remove this file it is not related to this ticket

File: src/components/createEditCqlLibrary/CreateEditCqlLibrary.tsx [hunk 2]
```diff
@@ -116,81 +117,99 @@ const CreateEditCqlLibrary = () => {
...
-          {serverError}
-        </ErrorAlert>
-      )}
-      <form
-        data-testid="create-new-cql-library-form"
-        onSubmit={formik.handleSubmit}
+    <>
+      <div
+        tw="flex flex-wrap "
+        className="flex justify-center md:justify-end"
```
Comment (adongare) [inline src/components/createEditCqlLibrary/CreateEditCqlLibrary.tsx]: looks like we are adding the classes twice

File: src/components/createEditCqlLibrary/CreateEditCqlLibrary.tsx [hunk 3]
```diff
@@ -116,81 +117,99 @@ const CreateEditCqlLibrary = () => {
...
+        style={{ marginBottom: "-5.7rem" }}
```
Comment (adongare) [inline src/components/createEditCqlLibrary/CreateEditCqlLibrary.tsx]: ~~can we not use class `mb-...`~~

File: src/components/createEditCqlLibrary/CreateEditCqlLibrary.tsx [hunk 4]
```diff
@@ -116,81 +117,99 @@ const CreateEditCqlLibrary = () => {
...
+                  })}
+                </TextField>
+              </FormControl>
+              <FormRow>
+                <Button
+                  id="saveBtn"
+                  buttonTitle={id ? "Update CQL Library" : "Create Cql Library"}
+                  type="submit"
+                  tw="mr-3"
+                  data-testid="create-new-cql-library-save-button"
```
Comment (adongare) [inline src/components/createEditCqlLibrary/CreateEditCqlLibrary.tsx]: could this be `cql-library-save-button` and for cancel `cql-library-cancel-button`, because this is save+update I think.

File: src/components/createEditCqlLibrary/CreateEditCqlLibrary.tsx (hunk 2 above)
Comment (sb-prateekkeerthi) [reply src/components/createEditCqlLibrary/CreateEditCqlLibrary.tsx]: Thanks, I'll remove it

File: src/components/createEditCqlLibrary/CreateEditCqlLibrary.tsx (hunk 4 above)
Comment (sb-prateekkeerthi) [reply src/components/createEditCqlLibrary/CreateEditCqlLibrary.tsx]: I agree with, I have changed the names

**Review by adongare — APPROVED** [review]
